cmd/handlers/main.go

```diff
@@ -136,7 +149,15 @@ func main() {
 			} else {
 
 				// Add block to blockchain
-				if err := blockchain.AddBlock(newBlock, constants.BlockchainFile, constants.MetadataTable); err != nil {
+				blockchainFile, err := os.OpenFile(constants.BlockchainFile, os.O_APPEND|os.O_WRONLY, 0644)
```

Comment:
Why is this being opened a second time?

Reply:
The file is opened with different flags here for the purpose of appending new data, whereas when the file is first opened it was simply for read only.

Reply:
The `(*FILE) seek` function within the `getBlockBy...` funcs does not work as intended when a file is opened with the flag `O_APPEND`. That's why I chose to open the files for `addBlock` and `getBlockBy...` separately.

Reply:
My concern is that the second time we open the file, we are opening it within a loop, so it will happen each time. Is it possible to close the file pointer the first time once we're done, then open a new one for appending, and just reuse that for the loop?

Reply:
It is certainly possible to reuse the same file pointer for appending. I was concerned about opening a file indefinitely.

Reply:
I don't feel there's anything wrong with opening the file indefinitely. Although I suppose it might become an issue if we want to read from it. We can keep it as is unless it becomes an issue later.